utility/src/main/java/org/oppia/android/util/caching/testing/FakeAssetRepository.kt

```diff
@@ -66,17 +67,42 @@ class FakeAssetRepository @Inject constructor(
   }
 
   private fun loadTextFile(assetName: String): String {
-    return trackedAssets.computeIfAbsent(assetName) {
-      prodImpl.loadTextFileFromLocalAssets(assetName)
-    } as? String ?: error("Asset doesn't exist: $assetName")
+    return if (Build.VERSION.SDK_INT >= Build.VERSION_CODES.N) {
+      trackedAssets.computeIfAbsent(assetName) {
+        prodImpl.loadTextFileFromLocalAssets(assetName)
+      } as? String ?: error("Asset doesn't exist: $assetName")
+    } else {
+      val textData = trackedAssets[assetName]
+      return if (textData != null) {
+        textData as? String ?: error("Asset doesn't exist: $assetName")
+      } else {
+        val dataFromFile = prodImpl.loadTextFileFromLocalAssets(assetName)
+        trackedAssets.putIfAbsent(assetName, dataFromFile)
+        dataFromFile as? String ?: error("Asset doesn't exist: $assetName")
+      }
+    }
```

Comment:
Instead of having nested else branches, you can optimize the code differently:
```suggestion
    } else {
      val textData = trackedAssets[assetName] ?: synchronized(trackedAssets) {
        trackedAssets[assetName] ?: run {
          val dataFromFile = prodImpl.loadTextFileFromLocalAssets(assetName)
          trackedAssets[assetName] = dataFromFile
          dataFromFile
        }
      }
      return textData as? String ?: error("Asset doesn't exist: $assetName")
    }
```